GooglePlayInstant/LoadingScreen/LoadingBar.cs

```diff
@@ -0,0 +1,174 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace GooglePlayInstant.LoadingScreen
+{
+    /// <summary>
+    /// Class that encapsulates the creation and update of a Loading Bar component in the editor and during the game's
+    /// runtime.
+    /// </summary>
+    public static class LoadingBar
+    {
+        // TODO: revisit these arbitrarily chosen values
+        /// <summary>
+        /// Percentage of the loading bar allocated to the the asset bundle downloading process.
+        /// </summary>
+        public const float AssetBundleDownloadMaxWidthPercentage = .8f;
+
+        /// <summary>
+        /// Percentage of the loading bar allocated to the the scene loading process.
+        /// </summary>
+        public const float SceneLoadingMaxWidthPercentage = 1 - AssetBundleDownloadMaxWidthPercentage;
+
+        // Loading bar fill's padding against the loading bar outline.
+        private const int LoadingBarFillPadding = 17;
+
+        private const int LoadingBarHeight = 30;
+
+        // Loading bar width as a percentage canvas object's automatic size
+        private const float LoadingBarWidthPercentage = .5f;
+
+        // Loading bar y axis placement as a percentage of canvas object's automatic y value
+        private const float LoadingBarYAxisPercentage = 2f;
+
+        // Names for the gameobject components
+        private const string LoadingBarGameObjectName = "Loading Bar";
+        private const string LoadingBarOutlineGameObjectName = "Loading Bar Outline";
+        private const string LoadingBarFillGameObjectName = "Loading Bar Fill";
+
+#if UNITY_EDITOR
+        // TODO: fix persistance of loading bar
+        /// <summary>
+        /// Creates a loading bar component on the specified loading screen game object's bottom half. Consists of
+        /// a white rounded border, with a colored loading bar fill in the middle.
+        /// </summary>
+        public static void AddLoadingScreenBarComponent(GameObject loadingScreenGameObject)
+        {
+            var loadingBarGameObject = new GameObject(LoadingBarGameObjectName);
+            loadingBarGameObject.AddComponent<RectTransform>();
+            loadingBarGameObject.transform.SetParent(loadingScreenGameObject.transform);
+
+            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
+
+            var loadingScreenRectTransform = loadingScreenGameObject.GetComponent<RectTransform>();
+
+            //Set the size of the loading bar
+            loadingBarRectTransform.sizeDelta =
+                new Vector2(loadingScreenRectTransform.sizeDelta.x * LoadingBarWidthPercentage,
```

Comment:
I'm assuming we're using loadingScreenRectTransform.sizeDelta.x as the 100%

Reply:
Sorta. I didn't think it aesthetically looked right for the bar to be from one end of the android screen to the other end, so the LoadingBarWidthPercentage is to give it padding on either side. Added a clarifying comment.